test/EIP20Gateway/approveBountyAmount.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const chai = require('chai');
+const sinon = require('sinon');
+const EIP20Gateway = require('../../src/ContractInteract/EIP20Gateway');
+const EIP20Token = require('../../src/ContractInteract/EIP20Token');
+const SpyAssert = require('../../test_utils/SpyAssert');
+const AssertAsync = require('../../test_utils/AssertAsync');
+const TestMosaic = require('../../test_utils/GetTestMosaic');
+const assert = chai.assert;
+
+describe('EIP20Gateway.approveBountyAmount()', () => {
+  let mosaic;
+  let bountyAmount;
+  let txOptions;
+  let gateway;
+
+  let spyGetBounty;
+  let spyGetEIP20BaseToken;
+  let mockEIP20BaseToken;
+  let spyApprove;
+  let spyCall;
+
+  const setup = () => {
+    mockEIP20BaseToken = sinon.mock(
+      new EIP20Token(
```

Comment:
Similar comment for using `sinon.createStubInstance(EIP20Token))`
You could search usage of `sinon.mock` across the repo, and decide to change.